Common++/header/PointerVector.h

```diff
@@ -121,19 +122,36 @@ namespace pcpp
 			m_Vector.clear();
 		}
 
+		/**
+		 * Adding a nullptr to the vector is not allowed.
+		 */
+		void pushBack(std::nullptr_t element, bool freeElementOnError = false) = delete;
+
 		/**
 		 * Add a new (pointer to an) element to the vector
 		 * @param[in] element A pointer to an element to assume ownership of.
+		 * @param[in] freeElementOnError If set to true, the element is freed if an exception is thrown during the push.
 		 * @throws std::invalid_argument The provided pointer is a nullptr.
 		 */
-		void pushBack(T* element)
+		void pushBack(T* element, bool freeElementOnError = false)
```

Comment:
I wonder if setting `freeElementOnError` default as `true` makes sense? Seems that most of the time we set it as `true`.

Reply:
Maybe, @seladb wanted it set to false [here](https://github.com/seladb/PcapPlusPlus/issues/1465#issuecomment-2211723443) tho

Reply:
I see, @seladb do you consider let it to be true?

Reply:
I thought about it more and changed my mind: https://github.com/seladb/PcapPlusPlus/issues/1465#issuecomment-2252113356

Reply:
Cool, it makes sense to me as well.